CalendarApp/GoogleCalendarService/Manager/EventsListBuilder.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Google.Apis.Calendar.v3.Data;
+
+namespace GoogleCalendarService.Manager
+{
+    public class EventsListBuilder
+    {
+        private FrequencyEnumaretion _frequencyEnumaretion;
+        private Event _realEvent;
+        private FrequencyType _freqType;
+        private int _count;
+        private DateTime _untilDateTime;
+        private List<DayOfWeek> _dayOfWeeksList;
+        private List<int> _dayJumpList; 
+
+        public List<Event> Build()
+        {
+            DateTime startTime = _realEvent.Start.DateTime.Value;
+            DateTime endTime = _realEvent.End.DateTime.Value;
+            var generatedEvents = new List<Event>();
+            int i = 0;
+
+            while(true)
+            {
+                generatedEvents.Add(new Event()
+                {
+                    Id = _realEvent.Id,
+                    Start = new EventDateTime()
+                    {
+                        DateTime = startTime,
+                    },
+                    End = new EventDateTime()
+                    {
+                        DateTime = endTime,
+                    }
+                });
+
+                switch (_frequencyEnumaretion)
+                {
+                    case FrequencyEnumaretion.DAILY:
+                        startTime = startTime.AddDays(1);
+                        endTime = endTime.AddDays(1);
+                        break;
+                    case FrequencyEnumaretion.WEEKLY:
+                        startTime = startTime.AddDays(_dayJumpList[i]);
+                        endTime = endTime.AddDays(_dayJumpList[i]);
+                        break;
+
+                    case FrequencyEnumaretion.MONTHLY:
+                        int day = startTime.Day;
+                        do
+                        {
+                            startTime = startTime.AddMonths(1);
+                            endTime = endTime.AddMonths(1);
+
+                        } while (day != startTime.Day);
+                        break;
+
+                    case FrequencyEnumaretion.YEARLY:
+                        startTime = startTime.AddYears(1);
+                        endTime = endTime.AddYears(1);
+                        break;
+                }
+
+                if (CheckForCrossBreak(_freqType, i, _count, startTime, _untilDateTime))
+                    break;
+                i++;
+            }
+
+            return generatedEvents;
+        }
```

Comment:
The idea of builder patter is that u can configure and then build any number of objects with same configuration by calling build()

So I'd move the logic form this method to other 'With' methods and would only return generated events here